client/telemetry/src/lib.rs

```diff
@@ -18,86 +18,50 @@
 
 //! Telemetry utilities.
 //!
-//! Calling `init_telemetry` registers a global `slog` logger using `slog_scope::set_global_logger`.
-//! After that, calling `slog_scope::with_logger` will return a logger that sends information to
-//! the telemetry endpoints. The `telemetry!` macro is a short-cut for calling
-//! `slog_scope::with_logger` followed with `slog_log!`.
+//! `Telemetry` objects can be created through its constructor `Telemetry::new()`, or through a
+//! `Telemetries` instance. The difference between the two is that `Telemetries` will re-use
+//! connections to the same server if possible and manages a collection of channel `Sender` for you
+//! (see `Senders`). `Telemetries` should be used unless you need finer control.
 //!
-//! Note that you are supposed to only ever use `telemetry!` and not `slog_scope::with_logger` at
-//! the moment. Substrate may eventually be reworked to get proper `slog` support, including sending
-//! information to the telemetry.
+//! The macro `telemetry!` can be used to report telemetries from anywhere but a `Telemetry` must
+//! have been initialized. Creating a `Telemetry` will make all the following code execution use
+//! this `Telemetry` when reporting with the macro `telemetry!` until the `Telemetry` object is
+//! dropped. If multiple `Telemetry` objects are created, the latest one (higher up in the stack)
```

Comment:
So that works similar to the prefixing layer? I'm a little bit confused.

Reply:
Yes that's it. If there are multiple span with telemetries, the one chosen is the "latest" one (because technically you're in 2 spans).

Reply:
So as for the prefixing layer, it means everything that is in the same scope, will use this span?

Reply:
Yes exactly